dnfdaemon-client/commands/remove/remove.cpp

```diff
@@ -0,0 +1,88 @@
+(at your option) any later version.
+
+Dnfdaemon-client is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with dnfdaemon-client.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "remove.hpp"
+
+#include "../../context.hpp"
+#include "../../utils.hpp"
+
+#include <dnfdaemon-server/dbus.hpp>
+#include <libdnf-cli/argument_parser.hpp>
+#include <libdnf/conf/option_bool.hpp>
+#include <libdnf/conf/option_string.hpp>
+
+#include <iostream>
+#include <memory>
+
+namespace dnfdaemon::client {
+
+void CmdRemove::set_argument_parser(Context & ctx) {
+    auto remove = ctx.arg_parser.add_new_command("remove");
+    remove->set_short_description("remove packages on the system");
+    remove->set_description("");
+    remove->set_named_args_help_header("Optional arguments:");
+    remove->set_positional_args_help_header("Positional arguments:");
+    remove->set_parse_hook_func([this, &ctx](
+                                     [[maybe_unused]] libdnf::cli::ArgumentParser::Argument * arg,
+                                     [[maybe_unused]] const char * option,
+                                     [[maybe_unused]] int argc,
+                                     [[maybe_unused]] const char * const argv[]) {
+        ctx.select_command(this);
+        return true;
+    });
+    ctx.arg_parser.get_root_command()->register_command(remove);
+
+    patterns_options = ctx.arg_parser.add_new_values();
+    auto keys = ctx.arg_parser.add_new_positional_arg(
+        "keys_to_match",
+        libdnf::cli::ArgumentParser::PositionalArg::UNLIMITED,
+        ctx.arg_parser.add_init_value(std::unique_ptr<libdnf::Option>(new libdnf::OptionString(nullptr))),
+        patterns_options);
+    keys->set_short_description("List of packages to remove");
+    remove->register_positional_arg(keys);
+}
+
+void CmdRemove::run(Context & ctx) {
+    if (!am_i_root()) {
+        std::cout << "This command has to be run with superuser privileges (under the root user on most systems)."
+                  << std::endl;
+        return;
+    }
+
+    // get package specs from command line and add them to the goal
+    std::vector<std::string> patterns;
+    if (patterns_options->size() > 0) {
+        patterns.reserve(patterns_options->size());
+        for (auto & pattern : *patterns_options) {
+            auto option = dynamic_cast<libdnf::OptionString *>(pattern.get());
+            patterns.emplace_back(option->get_value());
+        }
+    }
+
+    dnfdaemon::KeyValueMap options = {};
+
+    ctx.session_proxy->callMethod("remove")
+        .onInterface(dnfdaemon::INTERFACE_RPM)
+        .withTimeout(-1)
+        .withArguments(patterns, options);
+
+    run_transaction(ctx);
+}
```

Comment:
In DNF remove command results in allow_erasing = True. Dou you want to use the same logic?

Reply:
It is needed for removal of dependent packages, right?
I'll think about it. One possible solution is to override `allow_erasing` command-line argument value to be always `true` for `remove` command in the dnfdaemon-client.
Does it make sense to offer an API for package removal without allowerasing? If not then this should be handled on the server side so that API client does not need to specify `allowerasing=true` each time he is resolving a goal with package remova request.